### PR TITLE
`<DropdownLayout/>` - move options validation to propTypes

### PR DESCRIPTION
**Pull request by hepiyellow**
Resolves https://github.com/wix/wix-style-react/issues/2714

### CODE REVIEW

File: src/DropdownLayout/DropdownLayout.spec.js [hunk 1]
```diff
@@ -605,4 +606,55 @@ describe('DropdownLayout', () => {
...
+      });
+    });
+
+    describe('invalid', () => {
+      it('no value', () => {
+        render(<DropdownLayout options={[{ id: '1' }]} />);
+        expect(consoleErrors.get()).toHaveLength(1);
+        expect(consoleErrors.get()[0].includes('option.value')).toBeTruthy();
+
+        consoleErrors.reset(); // prevent test from failing
```
[the remark this answers is not in the document]
Comment (hepiyellow) [reply src/DropdownLayout/DropdownLayout.spec.js]: I think I can't, because then the test will fail

[the remark this answers is not in the document]
Comment (hepiyellow) [reply src/DropdownLayout/DropdownLayout.spec.js]: I used spyOn, and added before and after Each. Thanks.

File: src/DropdownLayout/DropdownLayout.js [hunk 2]
```diff
@@ -429,6 +423,45 @@ class DropdownLayout extends WixComponent {
   }
 }
 
+const optionPropTypes = PropTypes.shape({
```
[the remark this answers is not in the document]
Comment (hepiyellow) [reply src/DropdownLayout/DropdownLayout.js]: I realized that when using `PropTypes.onOf([A,B])`,
then if the validation fails, the message would be simply "Invalid prop...",
While if I have only `PropTypes.shape` then the message is more informative.

So I do the divider validation in code in the `optionValidator`.

I'll add comments for that.

File: src/DropdownLayout/DropdownLayout.js [hunk 3]
```diff
@@ -429,6 +423,45 @@ class DropdownLayout extends WixComponent {
...
+  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
+  value: PropTypes.oneOfType([PropTypes.node, PropTypes.string, PropTypes.func])
+    .isRequired,
+  disabled: PropTypes.bool,
+  overrideStyle: PropTypes.bool,
+});
+
+function optionValidator(props, propName, componentName) {
```
[the remark this answers is not in the document]
Comment (hepiyellow) [reply src/DropdownLayout/DropdownLayout.js]: sounds good.

[the remark this answers is not in the document]
Comment (yardnsm) [reply src/DropdownLayout/DropdownLayout.js]: Autodocs seems to fail to follow the reference, but I'm not sure exactly.

**Comment by yardnsm** [comment]
Also, are you going to remove `_isLegalOption`? It seems to be unused after the changes you've made.

**Comment by hepiyellow** [comment]
Ready to merge

**Pull request merged**